fetchlogs

```diff
@@ -0,0 +1,166 @@
+                create_text_file(os.path.join(current_container_logs_dir, '%s_error.txt' % container_subname),
+                                 error_message)
+        else:
+            # Copying logfile returned with `"docker inspect --format='{{.LogPath}}' %s" % container_name` does not work
+            # since: 1. it returns non-existing log filenames in some cases 2. sudo is mandatory to access file directly
+            try:
+                subprocess.check_output(
+                    'docker logs %s >& %s' % (container_name, os.path.join(temp_dir_path, container_subname)),
+                    shell=True
+                )
+            except exceptions.Exception as e:
+                error_message = 'unable to fetch logs of %s container with error:\n%s' % (container_name, e)
+                print(error_message)
+                create_text_file(os.path.join(temp_dir_path, '%s_error.txt' % container_subname), error_message)
+
+    if args.db_dump:
+        postgres_container_name = [cname for cname in running_containers if '_postgres_' in cname][0]
+        print('Creating Posgtres DBs dump - this may take up to several minutes...')
+        pgdump_filename = 'pgdump'
+        try:
+            subprocess.check_output(
+                'docker exec -i %s pg_dumpall -U postgres -f %s && '
+                'docker cp %s:/tmp/pgdump %s && '
+                'docker exec -i %s rm -rf /tmp/pgdump' % (
+                    postgres_container_name,
+                    os.path.join('/tmp', pgdump_filename),
+                    postgres_container_name,
+                    temp_dir_path,
+                    postgres_container_name
+                ),
+                shell=True
+            )
+        except exceptions.Exception as e:
+            error_message = 'unable to process the database dump with error:\n%s' % e
+            print(error_message)
+            create_text_file(os.path.join(temp_dir_path, '%s_error.txt' % pgdump_filename), error_message)
+    try:
+        print('Zipping data - almost done...')
+        zip_logs()
+    except exceptions.Exception as e:
+        print('unable to archive the directory %s into zip archive. Check the logs in directory: %s\nError:\n%s' % (
+            archive_output_dir,
+            temp_dir_path,
+            e
+        ))
+        sys.exit(8)
+    try:
+        shutil.rmtree(temp_dir_path)
+    except exceptions.Exception as e:
+        print('unable to clean the temporary directory (%s) with the error:\n%s' % (temp_dir_path, e))
+
+
+def create_text_file(file_path, text):
+    try:
+        error_msg_file = open(file_path, "w")
+        error_msg_file.write(text)
+        error_msg_file.close()
+    except exceptions.Exception as e:
+        print('SCRIPT ERROR: unable to create file %s with an error: %s' % (file_path, e))
+
+
+def zip_logs():
+    work_dir = os.getcwd()
+    os.chdir(archive_output_dir)
+    zipf = zipfile.ZipFile(output_zip_name, 'w', zipfile.ZIP_DEFLATED)
```

Comment:
output_zip_name should be an argument to `zip_logs`
Additionally, `ZipFile` has a context manager:
```
with zipfile.ZipFile(output_zip_name, 'w', zipfile.ZIP_DEFLATED) as zipf:
   ...
```